server/parameters/config.rs

```diff
@@ -41,23 +41,25 @@ impl Config {
     }
 
     pub fn new_with_encryption_config(encryption_config: EncryptionConfig, is_development_mode: bool) -> Self {
-        Self::customised(Some(encryption_config), None, None, is_development_mode)
+        Self::customised(None, Some(encryption_config), None, None, is_development_mode)
     }
 
     pub fn new_with_diagnostics_config(diagnostics_config: DiagnosticsConfig, is_development_mode: bool) -> Self {
-        Self::customised(None, Some(diagnostics_config), None, is_development_mode)
+        Self::customised(None, None, Some(diagnostics_config), None, is_development_mode)
     }
 
     pub fn new_with_data_directory(data_directory: &Path, is_development_mode: bool) -> Self {
-        Self::customised(None, None, Some(data_directory.to_path_buf()), is_development_mode)
+        Self::customised(None, None, None, Some(data_directory.to_path_buf()), is_development_mode)
```

Comment:
We do need to refactor it at some point. Somehow... It's getting out of control. Now a priority now.

Reply:
Yeah I've got an idea as to how it should look like